include/soci/db2/soci-db2.h

```diff
@@ -21,7 +21,11 @@
 //
 // If SOCI_DB2_DECL isn't defined yet define it now
 #ifndef SOCI_DB2_DECL
-# define SOCI_DB2_DECL
+# if __GNUC__ >= 4
+#  define SOCI_DB2_DECL __attribute__ ((visibility ("default")))
+# else
+#  define SOCI_DB2_DECL
+# endif
```

Comment:
We really must not repeat this in all backends, let's define `SOCI_DECL_{EXPORT,IMPORT}` once in `include/soci/soci-platform.h` and use them here.

Also, the definition itself is not correct: we must use DLL export/import declarations even when using gcc under MSW, as visibility is for ELF platforms only. In fact, I don't know if the tests for gcc version is sufficient, there definitely _used_ to be platforms where even much newer gcc versions had trouble with visibility, but maybe they don't exist any more. Ideal would be to use some CMake module detecting visibility support, there must be one.

Reply:
```c++
// If SOCI_DECL isn't defined yet define it now             
#ifndef SOCI_DECL                                           
# if __GNUC__ >= 4                                          
#  define SOCI_DECL __attribute__ ((visibility ("default")))
# else                                                      
#  define SOCI_DECL                                         
# endif                                                     
#endif                                                      
```

The `SOCI_DECL` would only be defined as `__attribute__ ((visibility ("default")))` when it is not defined, which means 
that `_WIN32` is not defined.

Thus I think it won't break building on windows.

Reply:
I really don't have any programming experience on windows and I know nothing about its export and import.

Should I define `SOCI_DECL_EXPORT` and `SOCI_DECL_IMPORT` to `__declspec(dllexport)`  and 
`__declspec(dllimport)` respectively on windows and define both of them to `__attribute__ ((visibility ("default")))` on
linux?

Reply:
@vadz I've pushed a few commits that resolves this.